### PR TITLE
Contributor Documentation Revisited

### PR DESCRIPTION
**Pull request by mjbrender**
One area we have yet to define is when and where we want users (including maintainers) to use the `type/rfc` label. This PR is intended to introduce the why on using a RFC, while also including a refactoring of some contribution documentation.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,19 +1,61 @@
...
+When **proposing a new approach**, understanding the use case is key. We organize these requests into a RFC process. RFCs must include:
+- A start and end date for comments
+- A set of required participants
+- A proposal that includes what is in and out of scope
+
+### What is an RFC?
+
+The Snap maintainers use RFCs, loosely based off the [EITF practice](https://www.ietf.org/rfc.html), to discuss proposed architectural and organizational improvements to the project. While most RFCs will originate from the maintainers, we welcome all users suggesting improvements through the RFC process.
+
+When **proposing a new approach to solving a challenge**, whether technical or organizational, open a request for comment (RFC). If you are ever unsure whether you should do so, open a regular issue and maintainers will help you groom it into a RFC.
```
Comment (kjlyon) [inline CONTRIBUTING.md]: Do people who are not maintainers have the ability to open an issue other than a `regular issue`? What is the difference between a regular issue and a RFC issue -- isn't it just that an RFC has the ``type/rfc`` tag. I thought only maintainers could add tags to issues.

Comment (mjbrender) [inline CONTRIBUTING.md]: You're correct - only maintainers can label. I mean that someone could open an issue without the requested details above (about scope, etc) if they're unsure of whether it's an RFC or not. Glad to reword if you think that's confusing.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,19 +1,61 @@
...
+- Expected results
+- Actual results
+
+When **requesting a feature**, context is key. Some of the questions we want to answer are:
+- What does this allow a user to do that they cannot do now?
+- What environment is this meant for (containerized, virtualized, bare metal)?
+- How urgent is the need (nice to have feature or need to have)?
+- Does this align with the goals of Snap?
+
+When **proposing a new approach**, understanding the use case is key. We organize these requests into a RFC process. RFCs must include:
```
Comment (kjlyon) [inline CONTRIBUTING.md]: ``RFCs must include...`` Change to: ```All Requests For Comment (RFC) documents must include...```

Comment (kjlyon) [inline CONTRIBUTING.md]: Re-reading that comment I'm not sure that is the best way to address this. I'm trying to think of a way to address the following questions seen from an external contributor or user's pov:
- What is an RFC, 
- How do I create one?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,19 +1,61 @@
...
+- A start and end date for comments
+- A set of required participants
```
Comment (kjlyon) [inline CONTRIBUTING.md]: Will external contributors (especially new people) know who the important people are to contact?

File: CONTRIBUTING.md (hunk 1 above)
Comment (kjlyon) [inline CONTRIBUTING.md]: This **proposing a new approach** section is already said above in the `RFCs must include` section

File: docs/BUILD_AND_TEST.md [hunk 4]
```diff
@@ -83,8 +76,22 @@ It should be noted here that if there are any untagged tests in the directory re
 
 Once the maintainers feel that the *small* tests provide sufficient code coverage, the existing *legacy* tests will be phased out (or used in the construction of a set of *medium* and *large* tests for the Snap CI/CD toolchain). All new tests being added to the Snap framework by contributors should be marked as either `small`, `medium`, or `large` tests, depending on their scope.
 
+### Building Effective Small Tests
```
Comment (kjlyon) [inline docs/BUILD_AND_TEST.md]: Do we need similar sections for building effective medium and large tests?

File: CONTRIBUTING.md (hunk 1 above)
Comment (mjbrender) [inline CONTRIBUTING.md]: Addressed this.

File: CONTRIBUTING.md (hunk 2 above)
Comment (mjbrender) [inline CONTRIBUTING.md]: Cleaned this up.

File: docs/BUILD_AND_TEST.md (hunk 4 above)
Comment (mjbrender) [inline docs/BUILD_AND_TEST.md]: In the longer run, yes. Out of scope for this PR but wonder if @nanliu or @kindermoumoute feels that should be opened as a new issue.

**Comment by mjbrender** [comment]
I'll squash these down after a thumbs up from you @kjlyon.

**Comment by candysmurf** [comment]
LGTM

**Comment by mjbrender** [comment]
You jumped the gun on this merge! 🙁  I hadn't squashed my commits yet. Remember to let the original maintainer of the PR merge when they're done. 

Not a big deal, but remember our workflow, please and thank you. cc @intelsdi-x/snap-maintainers

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -34,16 +34,14 @@ When **requesting a feature**, context is key. Some of the questions we want to
 - How urgent is the need (nice to have feature or need to have)?
 - Does this align with the goals of Snap?
 
-When **proposing a new approach**, understanding the use case is key. We organize these requests into a RFC process. RFCs must include:
+When **proposing a RFC**, understanding the use case is key. This type of issue includes:
```
Comment (kjlyon) [inline CONTRIBUTING.md]: `an RFC` sounds slightly more correct to me than `a RFC`, not sure which is correct though

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -34,16 +34,14 @@ When **requesting a feature**, context is key. Some of the questions we want to
...
-- A proposal that includes what is in and out of scope
+- A proposal with a straightforward explanation as well as what is in and out of scope
 
 ### What is an RFC?
 
-The Snap maintainers use RFCs, loosely based off the [EITF practice](https://www.ietf.org/rfc.html), to discuss proposed architectural and organizational improvements to the project. While most RFCs will originate from the maintainers, we welcome all users suggesting improvements through the RFC process.
-
-When **proposing a new approach to solving a challenge**, whether technical or organizational, open a request for comment (RFC). If you are ever unsure whether you should do so, open a regular issue and maintainers will help you groom it into a RFC.
+The Snap maintainers use RFCs, loosely based off the [EITF practice](https://www.ietf.org/rfc.html), to discuss proposed architectural and organizational improvements to the project. While most RFCs will originate from the maintainers, we welcome all users suggesting improvements through the RFC process. If you're not sure if you need to open one, you can always [discuss your idea on Slack](http://slack.snap-telemetry.io) beforehand.
```
Comment (kjlyon) [inline CONTRIBUTING.md]: There should be no commas in the first sentence.  I like how the rest of this paragraph was reworded. Looks good!

**Pull request merged**